### PR TITLE
Moved blockifier execution to a separate extension, moved related log…

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**
…ic to a related module

commit-id:ab7582b7

---

**Stack**:
- #1285
- #1284 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

**Comment by MaksymilianDemitraszek** [comment]
This PR introduces CallToBlockifier extension which executes contract calls to blockifier. 
It moves related logic to the extension module

File: crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/mod.rs [hunk 1]
```diff
@@ -0,0 +1,166 @@
+use std::marker::PhantomData;
```
Comment (MaksymilianDemitraszek) [inline crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/mod.rs:1]: This is the most important change in this PR.

**Review by drknzz — APPROVED** [review]

File: crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs
Comment (piotmag769) [inline crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs:1]: [nit] Why does forge runtime has its own dir with mod.rs and cheatable_starknet_runtime doesn't?

File: crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/mod.rs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    type Runtime = IORuntime<'a>;
+
+    fn override_system_call(
+        &mut self,
+        selector: DeprecatedSyscallSelector,
+        vm: &mut VirtualMachine,
+        extended_runtime: &mut IORuntime<'a>,
+    ) -> Result<SyscallHandlingResult, HintError> {
+        match selector {
+            DeprecatedSyscallSelector::CallContract => {
```
Comment (piotmag769) [inline crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/mod.rs:51]: We could add a comment that we do it this way to change the execution context (I guess mainly the vm)

**Review by piotmag769 — APPROVED** [review]

File: crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs
Comment (MaksymilianDemitraszek) [reply crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs:1]: because it has submodules

**Pull request merged**